mne/cov.py

```diff
@@ -1575,6 +1603,8 @@ def compute_whitener(noise_cov, info, picks=None, rank=None,
         If True, return the rank used to compute the whitener.
 
         .. versionadded:: 0.15
+    diag : bool
+        Use a diagonal approximation of the noise covariance.
```

Comment:
versionadded

Reply:
wait, why do we need this? Why not use `make_ad_hoc_cov` or `noise_cov.as_diag()`?

Reply:
it's used in whiten_evoked which also has a diag option. I think it's cleaner to do all kind of whitening in the same function.

Reply:
`whiten_evoked` shouldn't have the option, either :) Rather than propagate this error, I'd rather deprecate it, or do the logic inside `whiten_evoked`. Does it really save that much work to have this option?